app/src/full/java/com/celzero/bravedns/adapter/FirewallAppListAdapter.kt

```diff
@@ -99,6 +105,17 @@ class FirewallAppListAdapter(
                 val connStatus = FirewallManager.connectionStatus(appInfo.uid)
                 uiCtx {
                     b.firewallAppLabelTv.text = appInfo.appName
+                    if (appInfo.hasInternetPermission(appInfo.packageName, packageManager)) {
```

Comment:
Considering we are using `AppInfo` table to store all info and this is an adapter, I tried to reduce the calculations part as much as possible without making any changes to database.